packages/@aws-cdk/aws-iam/lib/grant.ts

```diff
@@ -160,7 +167,7 @@ export class Grant {
     const statement = new PolicyStatement({
       actions: options.actions,
       resources: (options.resourceSelfArns || options.resourceArns),
-      principals: [options.grantee!.grantPrincipal]
+      principals: [options.resourcePolicyPrincipal || options.grantee!.grantPrincipal]
```

Comment:
missing test for this

Reply:
We don't have separate tests in the IAM library for methods like `addToPrincipalAndResource` - they're exercised only through tests in the particular Construct Libraries that use them. I'd rather not add them as part of this change. With the KMS test added above, this is now covered as well.